src/Microsoft.SqlTools.ServiceLayer/DacFx/Contracts/DeploymentOptions.cs

```diff
@@ -2,192 +2,217 @@
 
-        public bool RunDeploymentPlanExecutors { get; set; }
+        public DeploymentOptionProperty<bool> RunDeploymentPlanExecutors { get; set; }
 
-        public bool RegisterDataTierApplication { get; set; }
+        public DeploymentOptionProperty<bool> RegisterDataTierApplication { get; set; }
 
-        public bool PopulateFilesOnFileGroups { get; set; }
+        public DeploymentOptionProperty<bool> PopulateFilesOnFileGroups { get; set; }
 
-        public bool NoAlterStatementsToChangeClrTypes { get; set; }
+        public DeploymentOptionProperty<bool> NoAlterStatementsToChangeClrTypes { get; set; }
 
-        public bool IncludeTransactionalScripts { get; set; }
+        public DeploymentOptionProperty<bool> IncludeTransactionalScripts { get; set; }
 
-        public bool IncludeCompositeObjects { get; set; }
+        public DeploymentOptionProperty<bool> IncludeCompositeObjects { get; set; }
 
-        public bool AllowUnsafeRowLevelSecurityDataMovement { get; set; }
+        public DeploymentOptionProperty<bool> AllowUnsafeRowLevelSecurityDataMovement { get; set; }
 
-        public bool IgnoreWithNocheckOnCheckConstraints { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreWithNocheckOnCheckConstraints { get; set; }
 
-        public bool IgnoreFillFactor { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreFillFactor { get; set; }
 
-        public bool IgnoreFileSize { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreFileSize { get; set; }
 
-        public bool IgnoreFilegroupPlacement { get; set; }
+        public DeploymentOptionProperty<bool> IgnoreFilegroupPlacement { get; set; }
 
-        public bool DoNotAlterReplicatedObjects { get; set; }
+        public DeploymentOptionProperty<bool> DoNotAlterReplicatedObjects { get; set; }
 
-        public bool DoNotAlterChangeDataCaptureObjects { get; set; }
+        public DeploymentOptionProperty<bool> DoNotAlterChangeDataCaptureObjects { get; set; }
 
-        public bool DisableAndReenableDdlTriggers { get; set; }
+        public DeploymentOptionProperty<bool> DisableAndReenableDdlTriggers { get; set; }
 
-        public bool DeployDatabaseInSingleUserMode { get; set; }
+        public DeploymentOptionProperty<bool> DeployDatabaseInSingleUserMode { get; set; }
 
-        public bool CreateNewDatabase { get; set; }
+        public DeploymentOptionProperty<bool> CreateNewDatabase { get; set; }
 
-        public bool CompareUsingTargetCollation { get; set; }
+        public DeploymentOptionProperty<bool> CompareUsingTargetCollation { get; set; }
 
-        public bool CommentOutSetVarDeclarations { get; set; }
+        public DeploymentOptionProperty<bool> CommentOutSetVarDeclarations { get; set; }
 
-        public int CommandTimeout { get; set; } = 120;
+        public DeploymentOptionProperty<int> CommandTimeout { get; set; } = new DeploymentOptionProperty<int>(120);
```

Comment:
please add comments for what these timeout values mean, like how setting LongRunningCommandTimeout to 0 means to wait infinitely